From 3e0a37aced52996d8100fce52aa7f7ae31cbf075 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Thu, 15 May 2014 17:26:30 +0300
Subject: [PATCH 25/78] igb: Enable random mac address

Enable random mac address in order to let the driver up
if eeprom values are incorrect.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 drivers/net/ethernet/intel/igb/igb_main.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/ethernet/intel/igb/igb_main.c b/drivers/net/ethernet/intel/igb/igb_main.c
index 9affd7c..d370dda 100644
--- a/drivers/net/ethernet/intel/igb/igb_main.c
+++ b/drivers/net/ethernet/intel/igb/igb_main.c
@@ -2503,6 +2503,10 @@ static int igb_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 		/* copy the MAC address out of the NVM */
 		if (hw->mac.ops.read_mac_addr(hw))
 			dev_err(&pdev->dev, "NVM Read Error\n");
+
+	if (!is_valid_ether_addr(hw->mac.addr)) {
+		dev_info(&pdev->dev, "Random MAC Address\n");
+		random_ether_addr(hw->mac.addr);
 	}
 
 	memcpy(netdev->dev_addr, hw->mac.addr, netdev->addr_len);
-- 
1.9.1

